### PR TITLE
produce to specific subject

### PR DESCRIPTION
**Pull request by daniel-davidd**

### CODE REVIEW

File: connect.go [hunk 1]
```diff
@@ -306,6 +309,10 @@ func (opts Options) connect() (*Conn, error) {
 	stationUpdatesSubsLock.Lock()
 	defer stationUpdatesSubsLock.Unlock()
 	c.stationUpdatesSubs = make(map[string]*stationUpdateSub)
+
+	stationFunctionsSubsLock.Lock()
```
Comment (idanasulinmemphis) [inline connect.go:313]: Why should we lock here?

File: producer.go [hunk 2]
```diff
@@ -62,6 +64,8 @@ type createProducerResp struct {
 	PartitionsUpdate        PartitionsUpdate `json:"partitions_update"`
 	SchemaVerseToDls        bool             `json:"schemaverse_to_dls"`
 	ClusterSendNotification bool             `json:"send_notification"`
+	StationVersion          int              `json:"station_version"`
+	StationFirstFunctions   map[int]int      `json:"station_first_functions"`
```
Comment (idanasulinmemphis) [inline producer.go:68]: StationPartitionsFirstFunctions

File: producer.go [hunk 3]
```diff
@@ -409,9 +420,24 @@ func (opts *ProduceOpts) produce(p *Producer) error {
...
 	}
 
+	var fullSubjectName string
+	if _, ok := p.conn.stationFunctionSubs[sn]; ok {
+		partitionNumber, err := strconv.Atoi(strings.Split(streamName, "$")[1])
+		if err != nil {
+			return memphisError(err)
+		}
+		if funcID, ok := p.conn.stationFunctionSubs[sn].FunctionsDetails.PartitionsFunctions[partitionNumber]; ok {
+			fullSubjectName = fmt.Sprintf("%v.%v", streamName, funcID)
```
Comment (idanasulinmemphis) [inline producer.go:430]: Streamname.functions.funcId

**Review by idanasulinmemphis — CHANGES_REQUESTED** [review]

File: connect.go (hunk 1 above)
Comment (daniel-davidd) [reply connect.go:313]: changed to a lock at the station level in order to modify the map without interfering producers

**Review by idanasulinmemphis — APPROVED** [review]

**Pull request merged**